lib/puppet/util/checksums.rb

```diff
@@ -83,6 +83,147 @@ def sha256lite_hex_length
     sha256_hex_length
   end
 
+  # Calculate a checksum using Digest::SHA384.
+  def sha384(content)
+    require 'digest/sha2'
+    Digest::SHA384.hexdigest(content)
+  end
+
+  def sha384?(string)
+    string =~ /^\h{96}$/
+  end
+
+  def sha384_file(filename, lite = false)
+    require 'digest/sha2'
+
+    digest = Digest::SHA384.new
+    checksum_file(digest, filename,  lite)
+  end
+
+  def sha384_stream(lite = false, &block)
+    require 'digest/sha2'
+    digest = Digest::SHA384.new
+    checksum_stream(digest, block, lite)
+  end
+
+  def sha384_hex_length
+    96
+  end
+
+  def sha384lite(content)
```

Comment:
The lite algorithms are kind of an abomination. I'd probably avoid adding them.

Reply:
Sure will remove them. I was not aware of the background and their intent (apart from what I could gather what they were doing)

Reply:
if you have large files with source => puppet:/// you'll want the lite version to prevent server and agents from doing large checksum computations. i've seen agents increasing the load on the systems noticeably because of checksumming and large files.

Reply:
The lite versions grab the first 512 characters. Is there a reason you want more than sha256lite? I'm not sure the others would be usefully differentiated from that.